linker/descriptors.go

```diff
@@ -1047,6 +1047,12 @@ func (f *fldDescriptor) Cardinality() protoreflect.Cardinality {
 	case descriptorpb.FieldDescriptorProto_LABEL_REQUIRED:
 		return protoreflect.Required
 	case descriptorpb.FieldDescriptorProto_LABEL_OPTIONAL:
+		if f.Syntax() == protoreflect.Editions {
```

Comment:
Might be worth adding a comment that explaining that `FieldDescriptorProto.label` is always set to `LABEL_OPTIONAL` for editions.